server/service/hosts.go

```diff
@@ -186,18 +196,40 @@ func deleteHostsEndpoint(ctx context.Context, request interface{}, svc fleet.Ser
 		StatusFilter: req.Filters.Status,
 		TeamFilter:   req.Filters.TeamID,
 	}
-	err := svc.DeleteHosts(ctx, req.IDs, listOpt, req.Filters.LabelID)
-	if err != nil {
-		return deleteHostsResponse{Err: err}, nil
+
+	// Since bulk deletes can take a long time, after DeleteHostsTimeout, we will return a 202 (Accepted) status code
+	// and allow the delete operation to proceed.
+	var err error
+	deleteDone := make(chan bool, 1)
+	ctx = context.WithoutCancel(ctx) // to make sure DB operations don't get killed after we return a 202
+	go func() {
+		err = svc.DeleteHosts(ctx, req.IDs, listOpt, req.Filters.LabelID)
+		deleteDone <- true
+	}()
```

Comment:
just a pass-by comment.

should we [queue a job ](https://github.com/fleetdm/fleet/blob/b8a84993b2a96c1886c7ba5596813e8efbadc7ca/server/worker/worker.go#L30) instead? they're not a full fledged queue solution but we can keep track of things if something happens.

alternatively/additionally, can we log the error?

I think it'll also help with the `deleteHostsSkipAuthorization` hack as all can happen in the service.

Reply:
Added logging.

I think queueing a job is overkill. We don't actually expect customers to delete large numbers of hosts. This is largely something dev/QA does.